app/views/apipie_dsl/apipie_dsls/help.html.erb

```diff
@@ -1,58 +1,340 @@
+
+<p><%= _('To customize template per rendering, various inputs can be defined. Inputs are of several types. Different template types support different input types.') %></p>
+<table class="table table-condensed">
+  <thead>
+  <tr>
+    <th><%= _('Input type') %></th>
+    <th><%= _('Description') %></th>
+  </tr>
+  </thead>
+  <tbody>
+  <tr>
+    <td><%= _('User input') %></td>
+    <td><%= _('User needs to specify the value when the template is rendered. A value type can also be specified, determining the filling mode. Plain value can be specified as an arbitrary string, Search value allows the autocompletion in the UI, Date expects a date value and UI provides a calendar helper. User input can have default value set. The value can be hidden with asterisks, which is useful for sensitive information. For Plani value type, multiple options can be filled in. During the rendering, user needs to pick one of the specified values.') %></td>
+  </tr>
+  <tr>
+    <td><%= _('Fact value') %></td>
+    <td><%= _('This input value is fed from specified fact for the host, which the template is being rendered for. This way the template can access e.g. the host cpus number.') %></td>
+  </tr>
+  <tr>
+    <td><%= _('Variable') %></td>
+    <td><%= _('This input value is fed from parameter values calculated for the host, which the template is being rendered for. This way the template can access a value of a global parameter, which is applied to every host. If that parameter would be overridden e.g. on host subnet level, this value would be fed to this input.') %></td>
+  </tr>
+  <tr>
+    <td><%= _('Puppet parameter') %></td>
+    <td><%= _('Similar to variable input type, however it loads the value for specific smart class parameter. User needs to specify the puppet class and the smart class parameter name when defining the input.') %></td>
+  </tr>
+  </tbody>
+</table>
+
+<p><%= _('All inputs defined for a signle template must have a have unique name, that is then used in ERB code. Inputs can optionally have a description, that is rendered as an inline help, when the template is rendered.') %></p>
+
+<p><%= (_('Once a template input is defined in the Inputs tab and the template is saved, it can be used in the ERB. To load value specified through this input, global macro <code>input</code> is used. The input name must be specified as the argument. E.g. the example below will load the value through input called cpus.').html_safe) %>
+<p>
+
+  <%= apipie_dsl_example("CPUs: " + apipie_erb_wrap("input('cpus')"), "CPUs: 8") %>
+
+<p><%= (_("While rendering the template in preview mode, the input either loads the value or uses a placeholder in case, the input is not ready for the rendering. If the template is rendered for real and the input is not ready, an error is raised. The readyness criteria differs per input type, e.g. for Fact inputs linked to host's fact with name \"cpu_count\" is ready when @host facts contains value for such fact.")) %> </p>
```

Comment:
` <p><%= (_("While rendering the template in preview mode, the input either loads the value or uses a placeholder in case, the input is not ready for the rendering. If the template is rendered for real and the input is not ready, an error is raised. The readyness criteria differs per input type, e.g. for Fact inputs linked to host's fact with name \"cpu_count\" is ready when @host facts contains value for such fact.")) %> </p>`

Please check this as I might have changed the meaning while cleaning up:

` <p><%= (_("While rendering the template in preview mode, the input either loads the value or uses a placeholder in case the input is not ready for rendering. If the template is rendered and the input is not ready, an error is thrown. The readiness criteria differs per input type, e.g. for Fact inputs linked to a host's fact with the name \"cpu_count\" , when @host contains a value for the fact, the input is ready.")) %> </p>`